pkg/controller/direct/registry/registry.go

```diff
@@ -107,6 +139,17 @@ func RegisterModelWithReconcileGate(gvk schema.GroupVersionKind, modelFn ModelFa
 	}
 }
 
+func IsDirectGVK(gvk schema.GroupVersionKind) bool {
```

Comment:
We don't want a `IsDirectGVK` function because version does not determine if a resource is Direct or not. You can use `IsDirectByGK` (below) instead.

Reply:
IsDirectGVK here is used to match for the scenarios that require version information. E.g. when we generate reference docs, we only generate it for v1beta1 resources. So version information will be necessary for the caller.

Reply:
Based on the scenario you described, can it be
``` 
if gvk.Version == "v1beta1" {
      if IsDirectByGK(gvk.GroupKind) {
         ...
      }
}
```

Reply:
Thanks for the suggestion! I updated the logic so that registry.go is kept untouched.